router/h2/src/main/scala/io/buoyant/router/h2/StreamStatsFilter.scala

```diff
@@ -1,38 +1,41 @@
 package io.buoyant.router.h2
 
-import com.twitter.finagle.{Status => _, _}
-import com.twitter.finagle.buoyant.h2.{Request, Response, Stream}
+import com.twitter.finagle._
+import com.twitter.finagle.buoyant.h2.{param => h2param, _}
+import com.twitter.finagle.service.ResponseClass.Successful
 import com.twitter.finagle.stats.StatsReceiver
 import com.twitter.util._
 
 object StreamStatsFilter {
   val role = Stack.Role("StreamStatsFilter")
   val module: Stackable[ServiceFactory[Request, Response]] =
-    new Stack.Module1[param.Stats, ServiceFactory[Request, Response]] {
-      override def role = StreamStatsFilter.role
+    new Stack.Module2[param.Stats, h2param.H2ResponseClassifier, ServiceFactory[Request, Response]] {
+      override def role: Stack.Role = StreamStatsFilter.role
       override def description = "Record stats on h2 streams"
-      override def make(statsP: param.Stats, next: ServiceFactory[Request, Response]) = {
+      override def make(
+        statsP: param.Stats,
+        classifierP: h2param.H2ResponseClassifier,
+        next: ServiceFactory[Request, Response]
+      ): ServiceFactory[Request, Response] = {
         val param.Stats(stats) = statsP
-        new StreamStatsFilter(stats).andThen(next)
+        val h2param.H2ResponseClassifier(classifier) = classifierP
+        new StreamStatsFilter(stats, classifier).andThen(next)
       }
     }
 }
 
-class StreamStatsFilter(statsReceiver: StatsReceiver)
+class StreamStatsFilter(statsReceiver: StatsReceiver, classifier: H2ResponseClassifier)
   extends SimpleFilter[Request, Response] {
 
-  class StreamStats(
-    stats: StatsReceiver,
-    durationName: Option[String] = None,
-    successName: Option[String] = None
-  ) {
-
+  // the `StreamStats` (and the related `FrameStats`) class is really just
+  // a "stateful function" – essentially a closure, but closing over state
```

Comment:
This is just my thought process out loud, I guess  – iteration one of `StreamStatsFilter` had all of the stats floating around as instance variables, so I was conceptualizing this as sort of being like a closure over them?